pythonPackages.pythonparser: init at 1.3

clacke opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

clacke committed:
pythonPackages.pythonparser: init at 1.3

worldofpeace commented on maintainers/maintainer-list.nix:
```diff
@@ -867,6 +867,11 @@
     github = "cko";
     name = "Christine Koppelt";
   };
+  clacke = {
```
Please add yourself as a maintainer in a separate commit.

Like `maintainers: add clacke`.

clacke replied:
✓

worldofpeace commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,30 @@
+    regex
+  ];
+
+  # Tests pass, but test cleanup is broken.
```
Please elaborate? Usually we can possibly fix this during review.

clacke replied:
✓Fixed breakage and enabled tests.

worldofpeace commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,30 @@
+    inherit pname version;
+    sha256 = "0qby6w2pcfwxi4w62rly08d9wj32amf6m7rmbrqw0xz6mirq7rrp";
+  };
+  LC_ALL = "en_US.UTF-8";
```
Was this because of a unicode issue? Also for this to work you need to add  ` glibcLocales` to `nativeBuildInputs`.

clacke replied:
✓ This was a copy-paste leftover. Removed.

worldofpeace reviewed (commented):
Additonally, a more preferred commit msg would be 
`pythonPackages.pythonparser: init at 1.3`

clacke committed:
pythonPackages.pythonparser: Python 3.7 not supported

A couple of tests fail with:

    NotImplementedError: pythonparser.lexer.Lexer cannot lex Python (3, 7)

Disable for Python 3.7 and above.

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -692,6 +692,8 @@ in {
 
   python-utils = callPackage ../development/python-modules/python-utils { };
 
+  pythonparser = disabledIf (pythonAtLeast "3.7") (callPackage ../development/python-modules/pythonparser { });
```
```suggestion
  pythonparser = callPackage ../development/python-modules/pythonparser { };
```
Where do you get this info from? You should instead add `disabled = ...` inside the expression.

clacke replied:
✓

dotlambda replied:
> I documented the reason in the commit message

Please add a comment in the expression as you will have to squash the new commits into the initial one.

clacke replied:
right, ok

dotlambda commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,30 @@
+
+  meta = {
+    description = "A Python parser intended for use in tooling";
+    homepage = https://m-labs.hk/pythonparser;
```
Please add
```
    downloadPage = https://github.com/m-labs/pythonparser;
```
because that one is really hard to find.

clacke replied:
✓

dotlambda commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,30 @@
+  };
+
+  patches = [
+    ./no_test_parser_teardown.patch
```
You could probably add `coverage` to `propagatedBuildInputs` instead.

clacke replied:
No, this is an internal grammar coverage module unrelated to the `coverage` package, and the failure is that it assumes a certain directory will exist.

I could create the directory instead of skipping the grammar coverage report, but the result won't end up in the package either way, the only difference would be `GRAMMAR COVERAGE: 99.93%` appearing in the log.

dotlambda commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,30 @@
+buildPythonPackage rec {
+  pname = "pythonparser";
+  version = "1.3";
+  src = fetchPypi {
```
Add a blank line above this.

clacke replied:
✓

dotlambda commented on pkgs/development/python-modules/pythonparser/default.nix:
```diff
@@ -0,0 +1,35 @@
+  pname = "pythonparser";
+  version = "1.3";
+
+  disabled = pythonAtLeast "3.7";
```
According to https://github.com/m-labs/pythonparser/issues/20, there's not even support for Python 3.6.

clacke replied:
Good find. Asking for clarification.

clacke replied:
On the other hand, as it builds and tests on Python 3.6, it still has value there, even though it might only be able to parse 3.5-compatible source. If someone tries to use it on sourcecode it doesn't understand, they will get errors, but if they don't, it can be installed together with some hypothetical 3.6-only package they might need.

dotlambda commented:
Please rebase on master now that you're already in `maintainer-list.nix`.

clacke commented:
Rebased, moved comments from commits to source, squashed, left the change still under discussion as a separate commit.

mmahut commented:
Are there any updates on this pull request, please?

FRidh commented:
Nothing happening so closing.
